### ISSUE TITLE
Question about dead case filling part of calculate_cases function

### ISSUE THREAD

**Issue by duinggulgi** [Neargye/magic_enum#368]
I have a question about dead case filling part of calculate_cases function.
Since the result array is not sorted, if we fill in the dead cases by increasing last_value (not max value) by 1,
There is a possibility of collision with non dead case hashes. 
Wouldn't it?

**Comment by Neargye** [comment]
You're right, the first dead-case strategy can theoretically collide with real hash values if the hashes of earlier enum entries happen to be larger than the last one. However, since all case labels are constexpr, any such collision would produce a compile-time error (duplicate case value), not silent wrong behavior at runtime. In practice this is extremely unlikely with the current hash function. We'll note this as a known edge case and may harden it in a future update.
